### PR TITLE
Implement CFA transfer based on grounds (reorganization, inheritance, court decision)

### PR DESCRIPTION
**Pull request by muzykantov**
- Added attributes to HLF: request_id, transfer basis, administrator_id, document_type, document_number, document_date, document_hashes
- Enabled transfer operations from wallets on the grey list
- Allowed transfer of funds blocked for external reasons by administrator
- Ensured transferred funds remain in "Blocked" status with new owner
- Restricted transfer of funds blocked for trading operations or investment requests
- Added "CFA Transfer Based on Grounds" section in the admin panel for asset transfers between users (CFA, GCP, fiat)
- Provided grounds for transfer: Reorganization (LE), Inheritance (IE), Court Decision (IE/LE)
- Admins with appropriate permissions can create, confirm/decline, and view transfer request details based on these grounds

### CODE REVIEW

File: core/reflectx/value_of.go [hunk 1]
```diff
@@ -57,35 +57,53 @@ func valueOf(s string, t reflect.Type) (reflect.Value, error) {
...
+//
+// Returns:
+//   - error: A formatted error message indicating the conversion failure.
+func errorValue(arg string, t reflect.Type, err error) error {
+	if err == nil {
+		return fmt.Errorf("%w: '%s': for type '%s'", ErrInvalidArgumentValue, arg, t.String())
 	}
 
-	return outValue, fmt.Errorf("%w: '%s': for type '%s'", ErrInvalidArgumentValue, s, t.String())
+	return fmt.Errorf("%w: '%s': for type '%s': '%s'", ErrInvalidArgumentValue, arg, t.String(), err.Error())
```
Comment (pfi79) [inline core/reflectx/value_of.go:108]: maybe `%w`?

Comment (muzykantov) [reply core/reflectx/value_of.go:108]: multiple wrapping works since go1.20 :(

File: core/reflectx/value_of.go [hunk 2]
```diff
@@ -58,34 +56,45 @@ func valueOf(s string, t reflect.Type) (reflect.Value, error) {
...
+// Returns:
+//   - error: A formatted error message indicating the conversion failure.
+func errorValue(arg string, t reflect.Type, errOrNil error) error {
+	err := fmt.Errorf("%w: '%s': for type '%s'", ErrInvalidArgumentValue, arg, t.String())
+	if errOrNil != nil {
+		err = fmt.Errorf("%v: '%w'", err, errOrNil)
 	}
 
-	return outValue, fmt.Errorf("%w: '%s': for type '%s'", ErrInvalidArgumentValue, s, t.String())
+	return err
```
Comment (pfi79) [inline core/reflectx/value_of.go:99]: I think it's supposed to be like this:

```
err := fmt.Errorf("%s: '%s' for type '%s': %w", ErrInvalidArgumentValue, "5", "int", errOrNil)
```

Comment (muzykantov) [reply core/reflectx/value_of.go]: Implemented wrapper. Probably it's not a best solution. But if you have any ideas about that - you're welcome.

File: core/external_transfers.go [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+		if !sender.Equal(admin) {
+			return ErrUnauthorisedNotAdmin
+		}
+	} else {
+		return fmt.Errorf("creating admin address: %w", err)
+	}
+
+	if req.GetRequestId() == "" {
+		req.RequestId = bc.stub.GetTxID()
+	}
```
Comment (pfi79) [inline core/external_transfers.go:35]: This field must always be filled in?

File: core/external_transfers.go [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+	fromAddress, err := types.AddrFromBase58Check(req.GetFromAddress())
+	if err != nil {
+		return fmt.Errorf("from address: %w", err)
+	}
+
+	toAddress, err := types.AddrFromBase58Check(req.GetToAddress())
```
Comment (pfi79) [inline core/external_transfers.go:42]: `fromAddress` and `toAddress` can be the same?

File: core/external_transfers.go (hunk 3 above)
Comment (muzykantov) [reply core/external_transfers.go:35]: Yup. Like in the locks method.

**Review by pfi79 — APPROVED** [review]

**Pull request merged**